modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,239 @@
+        def time_zone(request)
+          facility_id = request.dig(:facility, :parent_site_code)
+          facility = Mobile::VA_FACILITIES_BY_ID["dfn-#{facility_id}"]
+          facility ? facility[:time_zone] : nil
+        end
+
+        class VA
+          APPOINTMENT_TYPE = 'VA'
+
+          def self.provider_name(_)
+            nil
+          end
+
+          def self.practice_name(_)
+            nil
+          end
+
+          def self.facility_id(request)
+            Mobile::V0::Appointment.toggle_non_prod_id!(request.dig(:facility, :facility_code))
+          end
+
+          def self.location(request)
+            facility_id = facility_id(request)
+            facility = Mobile::VA_FACILITIES_BY_ID["dfn-#{facility_id}"]
+            {
+              id: facility_id,
+              name: facility ? facility[:name] : nil,
+              address: {
+                street: nil,
+                city: nil,
+                state: nil,
+                zip_code: nil
+              },
+              lat: nil,
+              long: nil,
+              phone: {
+                area_code: nil,
+                number: nil,
+                extension: nil
+              },
+              url: nil,
+              code: nil
+            }
+          end
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        class CC
+          APPOINTMENT_TYPE = 'COMMUNITY_CARE'
+
+          def self.provider_name(request)
+            provider_section = request.dig(:cc_appointment_request, :preferred_providers, 0)
+            if provider_section.nil? || (provider_section[:first_name].nil? && provider_section[:last_name].nil?)
+              return nil
+            end
+
+            "#{provider_section[:first_name]} #{provider_section[:last_name]}".strip
+          end
+
+          def self.practice_name(request)
+            request.dig(:cc_appointment_request, :preferred_providers, 0, :practice_name)
+          end
+
+          def self.facility_id(_)
+            nil
+          end
+
+          # rubocop:disable Metrics/MethodLength
+          def self.location(request)
+            source = request.dig(:cc_appointment_request, :preferred_providers, 0, :address) || {}
+            phone_captures = phone_captures(request)
+            {
+              id: nil,
+              name: practice_name(request),
+              address: {
+                street: source[:street],
+                city: source[:city],
+                state: source[:state],
+                zip_code: source[:zip_code]
+              },
+              lat: nil,
+              long: nil,
+              phone: {
+                area_code: phone_captures[1].presence,
+                number: phone_captures[2].presence,
+                extension: phone_captures[3].presence
+              },
+              url: nil,
+              code: nil
+            }
+          end
+          # rubocop:enable Metrics/MethodLength
+
+          def self.phone_captures(request)
```

Comment:
I cribbed this from one of our other adapters. If we find ourselves using it a third time, we should probably create a helper.

Reply:
Looks like we do it a third time in the modules/mobile/app/helpers/mobile/facilities_helper.rb:38. Regex is slightly different but I believe it's essentially equivalent. May be worth standardizing it with a helper. Given the scope of all these appointment additions maybe we break it out to a separate issue?

Reply:
Yeah, good idea.

Reply:
https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/department-of-veterans-affairs/va-mobile-app/2580